tonic/src/transport/endpoint.rs

```diff
@@ -252,3 +203,96 @@ impl fmt::Debug for Endpoint {
         f.debug_struct("Endpoint").finish()
     }
 }
+
+/// Configures TLS settings for endpoints.
+#[cfg(feature = "tls")]
+pub struct ClientTlsConfig {
+    provider: TlsProvider,
+    domain: Option<String>,
+    cert: Option<Certificate>,
+    #[cfg(feature = "openssl")]
+    openssl_raw: Option<openssl1::ssl::SslConnector>,
+    #[cfg(feature = "rustls")]
+    rustls_raw: Option<tokio_rustls::rustls::ClientConfig>,
+}
+
+#[cfg(feature = "tls")]
+impl fmt::Debug for ClientTlsConfig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ClientTlsConfig")
+            .field("provider", &self.provider)
+            .finish()
+    }
+}
+
+#[cfg(feature = "tls")]
+impl ClientTlsConfig {
+    /// Creates a new `ClientTlsConfig` backed by the specified provider. Enable the `openssl` or
+    /// `rustls` features of the `tonic` crate to use OpenSSL or Rustls respectively.
+    pub fn new(provider: TlsProvider) -> Self {
+        ClientTlsConfig {
+            provider,
+            domain: None,
+            cert: None,
+            #[cfg(feature = "openssl")]
+            openssl_raw: None,
+            #[cfg(feature = "rustls")]
+            rustls_raw: None,
+        }
+    }
+
+    /// Sets the domain name against which to verify the server's TLS certificate. If set to `None`
+    /// (the default), the address specified
```

Comment:
seems like the sentence got cut off?

Reply:
Yes, although actually it just occurred to me that there is no value in having this be an `Option` in any case now - if it doesn't need to be set, just omit the method call.

Reply:
I've made this take an `Into<String>` instead, and fixed the doc comment.